### ISSUE TITLE
EXOTransportRule - Some array properties are returned as empty even if a value is set

### ISSUE THREAD

**Issue by YenNantes** [microsoft/Microsoft365DSC#5575]
### Description of the issue

When using Export-M365DSCConfiguration with EXOTransportRule resources, some array properties show no value on the DSC result file while the get-transport rule cmdlet shows that the value is properly set.
I'm using the last M365DSC version and dependencies are up to date.
I have detected the issue at least on the following properties: ExceptIfHeaderContainsWords, FromAddressContainsWords, ExceptIfHeaderContainsWords.

### Microsoft 365 DSC Version

1.24.1211.1

### Which workloads are affected

Exchange Online

### The DSC configuration

```powershell
Export-M365DSCConfiguration -Components "EXOTransportRule"
```

### Verbose logs showing the problem

```powershell
For example for the policy below the "ExceptIfHeaderContainsWords" value (I have removed the other properties to keep it short:

DSC result file:
EXOTransportRule "EXOTransportRule-Email Tagging - Unverified Sender"
{
ExceptIfHeaderContainsWords = @();

  }
Get-TransportRule:
Get-TransportRule "Email Tagging - Unverified Sender" |select ExceptIfHeaderContainsWords

ExceptIfHeaderContainsWords
{1}
```

### Environment Information + PowerShell Version

```powershell

```

**Comment by YenNantes** [comment]
It seems that issue was introduced with version 1.24.1127.1.
probably  by this change:
![Image](https://github.com/user-attachments/assets/9aad4bb8-36e5-4c66-a998-c0dffed75584)

I have rolled back to version 1.24.1120.1 and it works well.

**Comment by jadamones** [comment]
I'm experiencing similar issues as well since 1.24.1127.1 #5500

**Comment by ykuijs** [comment]
This is indeed cause by [this PR](https://github.com/microsoft/Microsoft365DSC/pull/5462), which was released in 1.24.1127.1. Have also found a possible solution, but am now first checking what the rationale was of that earlier change.
